web/src/message_events.ts

```diff
@@ -85,13 +89,16 @@ export function update_current_view_for_topic_visibility() {
         // Use `set_timeout to call after we update the topic
         // visibility policy locally.
         // Calling this outside `user_topics_ui` to avoid circular imports.
+        assert(message_lists.current !== undefined);
         const msg_list_id = message_lists.current.id;
         setTimeout(() => {
+            assert(message_lists.current !== undefined);
             if (message_lists.current.id !== msg_list_id) {
                 // Check if the message list is still the same.
                 return;
             }
 
+            assert(filter !== undefined);
```

Comment:
Move this to the `if` so we can remove the `| undefined` from `filter_has_term_type`.